From 169d2b19aadf3eb6d9b5d973e1fb669d76d676ff Mon Sep 17 00:00:00 2001
From: tonyfan <waterwet@outlook.com>
Date: Sun, 18 Oct 2026 15:05:02 +0800
Subject: [PATCH] fix: honor device minimum refresh default

Settings and display policy must use the same device resource rather
than disagreeing on the initial minimum refresh rate.
---
 .../settings/display/MinRefreshRatePreferenceController.java  | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/com/android/settings/display/MinRefreshRatePreferenceController.java b/src/com/android/settings/display/MinRefreshRatePreferenceController.java
index b9c63040f26..472e30dc009 100644
--- a/src/com/android/settings/display/MinRefreshRatePreferenceController.java
+++ b/src/com/android/settings/display/MinRefreshRatePreferenceController.java
@@ -84,8 +84,10 @@ public class MinRefreshRatePreferenceController extends BasePreferenceController
 
     @Override
     public void updateState(Preference preference) {
+        final int configuredDefault = mContext.getResources().getInteger(
+                com.android.internal.R.integer.config_defaultMinRefreshRate);
         final float currentValue = Settings.System.getFloat(mContext.getContentResolver(),
-                MIN_REFRESH_RATE, 60.00f);
+                MIN_REFRESH_RATE, configuredDefault > 0 ? configuredDefault : 60.00f);
         int index = mListPreference.findIndexOfValue(
                 String.format(Locale.US, "%.02f", currentValue));
         if (index < 0) index = 0;
-- 
2.34.1

